### PR TITLE
Add support for Xata as a vector store

### PR DESCRIPTION
**Pull request by tsg**
This adds support for [Xata](https://xata.io) (data platform based on Postgres) as a vector store. We have recently added [Xata to Langchain.js](https://github.com/hwchase17/langchainjs/pull/2125) and would love to have the equivalent in the Python project as well.

The PR includes integration tests and a Jupyter notebook as docs. Please let me know if anything else would be needed or helpful.

I have added the xata python SDK as an optional dependency.

## To run the integration tests

You will need to create a DB in xata (see the docs), then run something like:

```
OPENAI_API_KEY=sk-... XATA_API_KEY=xau_... XATA_DB_URL='https://....xata.sh/db/langchain'  poetry run pytest tests/integration_tests/vectorstores/test_xata.py
```

### CODE REVIEW

**Comment by tsg** [comment]
@hwchase17 Thanks for approving the tests run and for the commit. I think I've solved the linting issue, but not sure if the right way, I've basically copied what I've seen in other files.

File: libs/langchain/langchain/vectorstores/xata.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+                    {"delete": {"table": self._table_name, "id": id}} for id in chunk
+                ]
+                self._client.records().transaction(payload={"operations": operations})
+        else:
+            raise ValueError("Either ids or delete_all must be set.")
+
+    def _delete_all(self) -> None:
+        """Delete all records in the table."""
+        while True:
+            r = self._client.search_and_filter().query(
```
Comment (philkra) [inline libs/langchain/langchain/vectorstores/xata.py:242]:
```suggestion
            r = self._client.data().query(
```

File: libs/langchain/langchain/vectorstores/xata.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+            ]
+            self._client.records().transaction(payload={"operations": operations})
+
+    def wait_for_indexing(self, timeout: float = 5, ndocs: int = 1) -> None:
+        """Wait for the search index to contain a certain number of
+        documents. Useful in tests.
+        """
+        start = time.time()
+        while True:
+            r = self._client.search_and_filter().search_table(
```
Comment (philkra) [inline libs/langchain/langchain/vectorstores/xata.py:261]:
```suggestion
            r = self._client.data().search_table(
```

File: libs/langchain/langchain/vectorstores/xata.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+        """
+        embedding = self._embedding.embed_query(query)
+        payload = {
+            "queryVector": embedding,
+            "column": "embedding",
+            "size": k,
+        }
+        if filter:
+            payload["filter"] = filter
+        r = self._client.search_and_filter().vector_search(
```
Comment (philkra) [inline libs/langchain/langchain/vectorstores/xata.py:187]:
```suggestion
        r = self._client.data().vector_search(
```

**Review by philkra — APPROVED** [review]
left one disregardable nit, LGTM

**Comment by tsg** [comment]
Thank you so much for the quick merge @baskaryan

**Pull request merged**